proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java

```diff
@@ -254,7 +261,21 @@ public void handle(Handshake handshake) throws Exception
         this.handshake = handshake;
         ch.setVersion( handshake.getProtocolVersion() );
 
-        // Starting with FML 1.8, a "\0FML\0" token is appended to the handshake. This interferes 
+        if ( handshake.getRequestedProtocol() == 2 && BungeeCord.getInstance().getConnectionThrottle().throttle( ( ( InetSocketAddress ) ch.getHandle().remoteAddress() ).getAddress() ) )
+        {
+            if ( BungeeCord.getInstance().getConfig().isLogThrottled() )
+            {
+                bungee.getLogger().log( Level.INFO, "{0} throttled", this );
+            }
+            // setting thisState to username to stop possible code execution on repeated handshakes
+            thisState = State.USERNAME;
```

Comment:
This is just equal to "accepting" the handshake as done in the switch below. I don't see your point here.

Or are you referring to the lines 272-273? I can include [FlexPipe patch 34](https://github.com/minotopiame/FlexPipe/blob/master/patches/0034-Ensure-kick-messages-can-be-sent-while-being-at-prot.patch) also if you want.

(The only reason this is at the top is to not execute the code until the switch.)